[Feature] [Beta] Add Aether Foundation, Team Skull, and Macro Cosmos as Evil Teams

Blitz425 opened the pull request:
## What are the changes?
Added Aether Foundation and Team Skull from Sun/Moon, and Macro Cosmos from Sword/Shield

## Why am I doing these changes?
Heavily requested feature from feature-vote in the discord, this additionally aims to add even more replayability and randomness factor to PokeRogue and give the player more references / love from the mainline titles to look forward to in their PokeRogue experience. 

## What did change?
- Added Aether/Skull/Macro as possible Evil Team Spawns
- Minor changes to a couple of the existing evil leader's teams to help redistribute to the new leaders.
- Minor changes to certain Grunt's pool of Pokemon to redistribute to others
### Screenshots/Videos
![image](https://github.com/user-attachments/assets/8619dc84-cea6-43c7-b84b-09e56669e5d0)
![image](https://github.com/user-attachments/assets/ce247386-a483-4b09-ab88-aa26f01077b1)
![image](https://github.com/user-attachments/assets/60039e04-97d3-40dc-87ae-fa579d230608)

## How to test the changes?
Start a classic run and check out the new encounters on floors 35, 62, 64, 66 ,112, 114, 115, and 165
You can also use overrides to jump to those floors

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

Tempo-anon commented on src/data/trainer-config.ts:
```diff
@@ -1192,7 +1198,7 @@ export const trainerConfigs: TrainerConfigs = {
       [TrainerPoolTier.UNCOMMON]: [Species.SOLROCK, Species.HIPPOPOTAS, Species.SANDACONDA, Species.PHANPY, Species.SWINUB, Species.GLIGAR],
       [TrainerPoolTier.RARE]: [Species.TRAPINCH, Species.HEATMOR],
-      [TrainerPoolTier.SUPER_RARE]: [Species.TURTONATOR, Species.CHARCADET]
+      [TrainerPoolTier.SUPER_RARE]: [Species.SCOVILLAIN, Species.CHARCADET]
```
I'm ok with this change but this should probably be documented in the PR desc just so all changes are accounted for

Tempo-anon commented on src/data/trainer-config.ts:
```diff
@@ -1222,6 +1228,27 @@ export const trainerConfigs: TrainerConfigs = {
       [TrainerPoolTier.RARE]: [Species.LITWICK, Species.SNEASEL, Species.PANCHAM, Species.PAWNIARD],
       [TrainerPoolTier.SUPER_RARE]: [Species.NOIVERN, Species.DRUDDIGON]
     }),
+  [TrainerType.AETHER_GRUNT]: new TrainerConfig(++t).setHasGenders("Aether Grunt Female").setHasDouble("Aether Foundation Employees").setMoneyMultiplier(1.0).setEncounterBgm(TrainerType.PLASMA_GRUNT).setBattleBgm("battle_plasma_grunt").setMixedBattleBgm("battle_aether_grunt").setVictoryBgm("victory_team_plasma").setPartyTemplateFunc(scene => getEvilGruntPartyTemplate(scene))
+    .setSpeciesPools({
+      [TrainerPoolTier.COMMON]: [ Species.PIKIPEK, Species.SNUBBULL, Species.POLIWAG, Species.ROCKRUFF, Species.LILLIPUP, Species.MAREANIE, Species.ALOLA_DIGLETT, Species.YUNGOOS, Species.STARYU, Species.CORSOLA],
+      [TrainerPoolTier.UNCOMMON]: [ Species.FOMANTIS, Species.ALOLA_SANDSHREW, Species.ALOLA_VULPIX, Species.ORICORIO, Species.STUFFUL, Species.BOUNSWEET, Species.MINIOR, Species.DEWPIDER, Species.ALOLA_GEODUDE, Species.TURTONATOR],
+      [TrainerPoolTier.RARE]: [ Species.ALOLA_EXEGGUTOR, Species.ORANGURU, Species.TOGEDEMARU, Species.PORYGON, Species.ALOLA_RAICHU, Species.PASSIMIAN, Species.GALAR_CORSOLA],
+      [TrainerPoolTier.SUPER_RARE]: [Species.JANGMO_O, Species.BAGON]
```
Aether is another one of those teams that's pretty creatively bankrupt huh. 

Grunt team comp looks fine to me, though are there any newly introduced gen 7 Pokemon that aren't being represented here?

Blitz425 replied:
Added a few more Pokemon from Gen 7 and removed a couple from other gens, there are maybe 1 or 2 species from Alola not covered but they just did not fit either Aether or Skull.

Tempo-anon commented on src/data/trainer-config.ts:
```diff
@@ -1222,6 +1228,27 @@ export const trainerConfigs: TrainerConfigs = {
+    }),
+  [TrainerType.SKULL_GRUNT]: new TrainerConfig(++t).setHasGenders("Skull Grunt Female").setHasDouble("Team Skull Grunts").setMoneyMultiplier(1.0).setEncounterBgm(TrainerType.PLASMA_GRUNT).setBattleBgm("battle_plasma_grunt").setMixedBattleBgm("battle_skull_grunt").setVictoryBgm("victory_team_plasma").setPartyTemplateFunc(scene => getEvilGruntPartyTemplate(scene))
+    .setSpeciesPools({
+      [TrainerPoolTier.COMMON]: [ Species.SALANDIT, Species.ALOLA_RATTATA, Species.SCRAGGY, Species.KOFFING, Species.ALOLA_GRIMER, Species.MAREANIE, Species.YUNGOOS, Species.SPINARAK],
```
I feel like Skull can use more bugs because Guzzma seems to be a bug type specialist.

Blitz425 replied:
Added a few more to them 👍

Tempo-anon commented on src/data/trainer-config.ts:
```diff
@@ -1222,6 +1228,27 @@ export const trainerConfigs: TrainerConfigs = {
+      [TrainerPoolTier.COMMON]: [ Species.SALANDIT, Species.ALOLA_RATTATA, Species.SCRAGGY, Species.KOFFING, Species.ALOLA_GRIMER, Species.MAREANIE, Species.YUNGOOS, Species.SPINARAK],
+      [TrainerPoolTier.UNCOMMON]: [ Species.FOMANTIS, Species.SABLEYE, Species.SANDILE, Species.ZUBAT, Species.HOUNDOUR, Species.MURKROW, Species.TRUBBISH, Species.ALOLA_MEOWTH, Species.DROWZEE],
+      [TrainerPoolTier.RARE]: [Species.ALOLA_MAROWAK, Species.SANDYGAST, Species.PAWNIARD, Species.MIMIKYU, Species.DHELMISE, Species.GASTLY, Species.WISHIWASHI, Species.PANCHAM],
+      [TrainerPoolTier.SUPER_RARE]: [Species.GALAR_SLOWPOKE, Species.MANKEY]
```
Super rare mankey and gslowpoke?

Blitz425 replied:
There's not much from their gen that makes sense on them and would also be rare worthy, originally had drampa as an option. I felt like Annihilape fit their aesthetic and general vibes (and one of the grunts did have a Primeape) and Galar Slowpoke is something goofy looking that they would have while also fitting the Poison type themeing they have.

Tempo-anon commented on src/data/trainer-config.ts:
```diff
@@ -1611,9 +1638,9 @@ export const trainerConfigs: TrainerConfigs = {
       p.abilityIndex = 2; // Swift Swim
     }))
-    .setPartyMemberFunc(3, getRandomPartyMemberFunc([ Species.HUNTAIL, Species.GOREBYSS ], TrainerSlot.TRAINER, true, p => {
+    .setPartyMemberFunc(3, getRandomPartyMemberFunc([ Species.OVERQWIL ], TrainerSlot.TRAINER, true, p => {
```
The gen 3 representation 😭

Tempo-anon commented on src/data/trainer-config.ts:
```diff
@@ -1628,9 +1655,9 @@ export const trainerConfigs: TrainerConfigs = {
     })),
   [TrainerType.CYRUS]: new TrainerConfig(++t).setName("Cyrus").initForEvilTeamLeader("Galactic Boss",[]).setMixedBattleBgm("battle_galactic_boss").setVictoryBgm("victory_team_plasma")
-    .setPartyMemberFunc(0, getRandomPartyMemberFunc([ Species.GYARADOS, Species.BASCULEGION ]))
+    .setPartyMemberFunc(0, getRandomPartyMemberFunc([ Species.GYARADOS ]))
```
Any reason for removing Bascuegion which I think is a fitting thematic variation for Gyarados?

Tempo-anon commented on src/data/trainer-config.ts:
```diff
@@ -1678,18 +1705,20 @@ export const trainerConfigs: TrainerConfigs = {
+    .setPartyMemberFunc(2, getRandomPartyMemberFunc([ Species.BASCULEGION ]))
     .setPartyMemberFunc(3, getRandomPartyMemberFunc([ Species.KINGAMBIT ]))
-    .setPartyMemberFunc(4, getRandomPartyMemberFunc([ Species.HYDREIGON, Species.IRON_JUGULIS ], TrainerSlot.TRAINER, true, p => {
+    .setPartyMemberFunc(4, getRandomPartyMemberFunc([ Species.GENESECT ], TrainerSlot.TRAINER, true, p => {
```
Why does Ghetsis2 have a shiny Genesect. I think this is a burn instead of a cook

Blitz425 replied:
Movie reference, was requested by a few people.

Tempo-anon commented on src/data/trainer-config.ts:
```diff
@@ -1678,18 +1705,20 @@ export const trainerConfigs: TrainerConfigs = {
     }))
-    .setPartyMemberFunc(1, getRandomPartyMemberFunc([ Species.DURANT ]))
-    .setPartyMemberFunc(2, getRandomPartyMemberFunc([ Species.DARMANITAN, Species.GALAR_DARMANITAN ]))
+    .setPartyMemberFunc(1, getRandomPartyMemberFunc([ Species.VOLCARONA ]))
+    .setPartyMemberFunc(2, getRandomPartyMemberFunc([ Species.BASCULEGION ]))
```
I can see Basculegion as something since it's an evo of Basculin, but I wonder if you can think of some variations here

Tempo-anon commented on src/data/trainer-config.ts:
```diff
@@ -1700,34 +1729,139 @@ export const trainerConfigs: TrainerConfigs = {
     .setPartyMemberFunc(2, getRandomPartyMemberFunc([ Species.PYROAR ]))
-    .setPartyMemberFunc(3, getRandomPartyMemberFunc([ Species.MILOTIC ]))
-    .setPartyMemberFunc(4, getRandomPartyMemberFunc([ Species.HELIOLISK ]))
+    .setPartyMemberFunc(3, getRandomPartyMemberFunc([ Species.AURORUS, Species.TYRANTRUM ]))
```
I get the variation thing here but this seems more in line with Diantha who actually used both.

Blitz425 replied:
Very fair, I just felt Heliolisk was an underwhelming pick but it could definitely be better

Tempo-anon commented on src/data/trainer-config.ts:
```diff
@@ -1700,34 +1729,139 @@ export const trainerConfigs: TrainerConfigs = {
       p.generateAndPopulateMoveset();
       p.pokeball = PokeballType.ULTRA_BALL;
       p.formIndex = 1;
+      p.shiny = true;
```
This is anime exclusive, I don't think this is necessary.

Blitz425 replied:
I feel like his shiny Gyarados is too notorious to pass up but this was also a request I got from people.

Tempo-anon commented on src/data/trainer-config.ts:
```diff
@@ -1700,34 +1729,139 @@ export const trainerConfigs: TrainerConfigs = {
     })),
   [TrainerType.LYSANDRE_2]: new TrainerConfig(++t).setName("Lysandre").initForEvilTeamLeader("Flare Boss",[], true).setMixedBattleBgm("battle_flare_boss").setVictoryBgm("victory_team_plasma")
-    .setPartyMemberFunc(0, getRandomPartyMemberFunc([ Species.SCREAM_TAIL, Species.FLUTTER_MANE ], TrainerSlot.TRAINER, true, p => {
+    .setPartyMemberFunc(0, getRandomPartyMemberFunc([ Species.ETERNAL_FLOETTE ], TrainerSlot.TRAINER, true, p => {
```
Not sure if this is a cook

Tempo-anon commented on src/data/trainer-config.ts:
```diff
@@ -1700,34 +1729,139 @@ export const trainerConfigs: TrainerConfigs = {
-    .setPartyMemberFunc(1, getRandomPartyMemberFunc([ Species.GHOLDENGO, Species.AEGISLASH ]))
-    .setPartyMemberFunc(2, getRandomPartyMemberFunc([ Species.PYROAR ]))
+    .setPartyMemberFunc(1, getRandomPartyMemberFunc([ Species.PYROAR ]))
+    .setPartyMemberFunc(2, getRandomPartyMemberFunc([ Species.IRON_MOTH ]))
```
Not really sure if I feel this one

Tempo-anon commented on src/data/trainer-config.ts:
```diff
@@ -1700,34 +1729,139 @@ export const trainerConfigs: TrainerConfigs = {
+    .setPartyMemberFunc(1, getRandomPartyMemberFunc([ Species.LILLIGANT, Species.HISUI_LILLIGANT ]))
+    .setPartyMemberFunc(2, getRandomPartyMemberFunc([ Species.MISMAGIUS, Species.WIGGLYTUFF ]))
+    .setPartyMemberFunc(3, getRandomPartyMemberFunc([ Species.ESPATHRA ]))
+    .setPartyMemberFunc(4, getRandomPartyMemberFunc([ Species.LOPUNNY ], TrainerSlot.TRAINER, true, p => {
+      p.generateAndPopulateMoveset();
+      p.formIndex = 1;
+    }))
+    .setPartyMemberFunc(5, getRandomPartyMemberFunc([ Species.CLEFABLE ], TrainerSlot.TRAINER, true, p => {
+      p.setBoss(true, 2);
+      p.generateAndPopulateMoveset();
+    })),
```
Clefable boss instead of Mega Lopunny? I see you're trying to make Mega Lopunny her "mega" but that really cuts into Bewear which I think is quite iconic for her and gen 7

Tempo-anon commented on src/data/trainer-config.ts:
```diff
@@ -1700,34 +1729,139 @@ export const trainerConfigs: TrainerConfigs = {
       p.pokeball = PokeballType.MASTER_BALL;
     })),
+  [TrainerType.LUSAMINE]: new TrainerConfig(++t).setName("Lusamine").initForEvilTeamLeader("Aether Boss",[]).setMixedBattleBgm("battle_aether_boss").setVictoryBgm("victory_team_plasma")
+    .setPartyMemberFunc(0, getRandomPartyMemberFunc([ Species.MILOTIC ]))
```
Would Pyukumuku work as a variant here since it's the same type and is from her generation and she has a frozen one?

Blitz425 replied:
not at all, Milotic is an actual mon and Pyukumuku is a worse Wobbuffet

Tempo-anon commented on src/data/trainer-config.ts:
```diff
@@ -1700,34 +1729,139 @@ export const trainerConfigs: TrainerConfigs = {
+    .setPartyMemberFunc(0, getRandomPartyMemberFunc([ Species.MILOTIC ]))
+    .setPartyMemberFunc(1, getRandomPartyMemberFunc([ Species.LILLIGANT, Species.HISUI_LILLIGANT ]))
+    .setPartyMemberFunc(2, getRandomPartyMemberFunc([ Species.MISMAGIUS, Species.WIGGLYTUFF ]))
+    .setPartyMemberFunc(3, getRandomPartyMemberFunc([ Species.ESPATHRA ]))
```
This one is sort of randomly picked. Was it chosen on purpose for type coverage or something?

Tempo-anon commented on src/data/trainer-config.ts:
```diff
@@ -1700,34 +1729,139 @@ export const trainerConfigs: TrainerConfigs = {
+      p.generateAndPopulateMoveset();
+      p.formIndex = 1;
+    }))
+    .setPartyMemberFunc(4, getRandomPartyMemberFunc([ Species.MAGEARNA ], TrainerSlot.TRAINER, true, p => {
```
Lusamine has a lot of powerful Pokemon she's canonically owned (Silvally, Celesteela, Naganadel) so I don't see how Magearna fits when it's from a different gen

Blitz425 replied:
Magearna is from Gen 7 actually and Lusamine canonically owns one (at least in the anime). Silvally is with Gladion and I don't remember her owning a Celesteela or Naganadel since they're given to the player.

Tempo-anon replied:
Couldn't find Magearna at all on her bulbapedia page https://bulbapedia.bulbagarden.net/wiki/Lusamine_(anime) but it did note her ultra beasts: https://bulbapedia.bulbagarden.net/wiki/Lusamine

H-A-R-V replied:
It is Mohn's her missing husband's. https://bulbapedia.bulbagarden.net/wiki/Mohn#Pok%C3%A9mon
It is also shiny.

Tempo-anon commented on src/data/trainer-config.ts:
```diff
@@ -1700,34 +1729,139 @@ export const trainerConfigs: TrainerConfigs = {
+      p.setBoss(true, 2);
+      p.generateAndPopulateMoveset();
+    }))
+    .setPartyMemberFunc(1, getRandomPartyMemberFunc([ Species.MILOTIC ]))
```
Is Milotic a hard Pokemon for trainers to deal with or something? You're already carrying over Clefable and Mega Lopunny so I don't know if you really need to carry over a third one esp with no variation

Blitz425 replied:
Eh I couldn't really find a good replacement for it, first thought was Primarina but that makes her even more poison weak. I left it just because I do think it's a really solid pokemon and provides her general type coverage but I'm open to suggestions

Tempo-anon commented on src/data/trainer-config.ts:
```diff
@@ -1700,34 +1729,139 @@ export const trainerConfigs: TrainerConfigs = {
+      p.pokeball = PokeballType.MASTER_BALL;
+    })),
+  [TrainerType.GUZMA]: new TrainerConfig(++t).setName("Guzma").initForEvilTeamLeader("Skull Boss",[]).setMixedBattleBgm("battle_skull_boss").setVictoryBgm("victory_team_plasma")
+    .setPartyMemberFunc(0, getRandomPartyMemberFunc([ Species.GOLISOPOD ], TrainerSlot.TRAINER, true, p => {
```
Keep in mind that the actual numbered party slots of the Pokemon in the party _will_ actually affect their levels, hence why I've been putting all the bosses in the last few slots

Blitz425 replied:
Was done intentionally for Emergency Exit + Regenerator Passive

Tempo-anon commented on src/data/trainer-config.ts:
```diff
@@ -1700,34 +1729,139 @@ export const trainerConfigs: TrainerConfigs = {
+      p.setBoss(true, 2);
+      p.generateAndPopulateMoveset();
+    }))
+    .setPartyMemberFunc(1, getRandomPartyMemberFunc([ Species.MASQUERAIN ]))
```
Masquerain sorta sucks i feel. I'd probably put Ariados as a possible alternative since that's more accurate to his in game team

Tempo-anon commented on src/data/trainer-config.ts:
```diff
@@ -1700,34 +1729,139 @@ export const trainerConfigs: TrainerConfigs = {
+      p.formIndex = 1;
+    })),
+  [TrainerType.ROSE]: new TrainerConfig(++t).setName("Rose").initForEvilTeamLeader("Macro Boss",[]).setMixedBattleBgm("battle_macro_boss").setVictoryBgm("victory_team_plasma")
+    .setPartyMemberFunc(0, getRandomPartyMemberFunc([ Species.ARCHALUDON ]))
+    .setPartyMemberFunc(1, getRandomPartyMemberFunc([ Species.FERROTHORN ]))
+    .setPartyMemberFunc(2, getRandomPartyMemberFunc([ Species.TINKATON ]))
+    .setPartyMemberFunc(3, getRandomPartyMemberFunc([ Species.CORVIKNIGHT, Species.SKARMORY ]))
+    .setPartyMemberFunc(4, getRandomPartyMemberFunc([ Species.PERRSERKER ]))
+    .setPartyMemberFunc(5, getRandomPartyMemberFunc([ Species.COPPERAJAH ], TrainerSlot.TRAINER, true, p => {
```
All these selections without any variation is sadge. If you're going to give Corviknight an alternative with Skarmory you can also consider Mawile for TInkaton and Klinklang for Perrserker

Tempo-anon commented on src/data/trainer-config.ts:
```diff
@@ -1700,34 +1729,139 @@ export const trainerConfigs: TrainerConfigs = {
+      p.generateAndPopulateMoveset();
+      p.pokeball = PokeballType.MASTER_BALL;
+    }))
+    .setPartyMemberFunc(4, getRandomPartyMemberFunc([ Species.MELMETAL ], TrainerSlot.TRAINER, true, p => {
+      p.setBoss(true, 2);
+      p.generateAndPopulateMoveset();
+      p.pokeball = PokeballType.MASTER_BALL;
+      p.formIndex = 1;
+    }))
+    .setPartyMemberFunc(5, getRandomPartyMemberFunc([ Species.COPPERAJAH ], TrainerSlot.TRAINER, true, p => {
+      p.setBoss(true, 2);
+      p.generateAndPopulateMoveset();
+      p.formIndex = 1;
```
Gee billy why do you get two g-maxes

Tempo-anon commented on src/data/trainer-config.ts:
```diff
@@ -1700,34 +1729,139 @@ export const trainerConfigs: TrainerConfigs = {
+    }))
+    .setPartyMemberFunc(1, getRandomPartyMemberFunc([ Species.EMPOLEON ]))
+    .setPartyMemberFunc(2, getRandomPartyMemberFunc([ Species.AEGISLASH, Species.GHOLDENGO ]))
+    .setPartyMemberFunc(3, getRandomPartyMemberFunc([ Species.IRON_CROWN ], TrainerSlot.TRAINER, true, p => {
+      p.generateAndPopulateMoveset();
+      p.pokeball = PokeballType.MASTER_BALL;
```
This can be variant'd with Cobalion and doesn't need a master ball

Blitz425 replied:
I meant to put Master Balls on any type of sublegend but forgot to apply it to others, I remember you mentioning how people liked pokeball types on Evil Teams so its a matter of how you would like to address that. Felt like Master Balls just made sense on those type of Pokemon.

Tempo-anon commented on src/data/trainer-config.ts:
```diff
@@ -1700,34 +1729,139 @@ export const trainerConfigs: TrainerConfigs = {
+      p.setBoss(true, 2);
+      p.generateAndPopulateMoveset();
+    }))
+    .setPartyMemberFunc(1, getRandomPartyMemberFunc([ Species.EMPOLEON ]))
```
Just Empoleon?

Blitz425 replied:
I felt like it fit his narcissistic rich guy attitude while also providing an answer to fire types, I think he has a lot of variation as is so there wouldn't be a need to add a variant but it would also have to provide some form of coverage.

Tempo-anon reviewed (commented):
1. You need to add `p.generateName()` for all Pokemon with alternate forms
2. Where slot on the team a Pokemon is affects its levels so unless you want to tweak the function to make it totally custom it'll follow existing structures. See: https://github.com/pagefaultgames/pokerogue/pull/2580
3. Comments about the team selection

Blitz425 commented:
> 3\. Comments about the team selection
When doing the teams I was following the general formula you did for the other Evil Leaders, 1 Mega, 1 Paradox/Sub Legend, 1 Legend (or another Sub Legend/Mythical if a fitting legend wasn't found.). 

Fight 1 mainly uses their in game team with some variation or something fitting from a future installment to "modernize" the teams as you did them with the other Evil Leaders. I redistributed some Pokemon from the other Leaders to distribute them to these new ones as I personally felt they fit the new leaders significantly more. 

Lusamine
Clefable was made the boss instead of her Mega Lopunny as it is her ace (and I'm following the logic you used with Cyrus and Weavile). There was discussion on what her mega should be, it was a choice between either Mega Lopunny or Mega Altaria which is a bit of a stretch since she gives one to Lillie in the anime. Lysandre lost Scream Tail / Flutter Mane to give to Lusamine as she also has the same warped perceptions of beauty (reasoning you gave for Lysandre) on top of the fact she actually has a Mismagius on her team and a Clefable that you could somewhat make a connection to Scream Tail for. I chose Eternal Floette as their replacement as it is also a Mythical Pokemon that retains the general fairy typing the paradoxes had while also being a really nice reference to XY and cool factor (the passive bonus of being a boss is also really nice for it).  Espathra was given to modernize the team in fight 1 but it felt too fitting to not give to her in just pure design but also it was nice type coverage.  She canonically owns a Magearna in the anime so it was chosen as the "boss" slot (Necrozma was intentionally not chosen for the same reason we can't give Rose Eternatus) and Nihilego is an obvious pick for her. 

Guzma
Team 1 is mainly his in game team, Heracross was added to keep the type coverage when going into Team 2 with his Buzzwole. Galvantula and Vikavolt are for the same reason turning into Xurkitree. Kleavor was made a variation over his Scizor. Golisopod was chosen as the boss in round 1 as it is his ace and intentionally made to start first as it pairs with Emergency Exit + Regenerator passive. Ariados was added as a variation on Masquerain per request. (Typing this I realize he doesn't have an extra sub legend to make up for his lack of legend, could maybe give him Slither Wing over the Scizor slot) Xurkitree was chosen as the "boss" as it is designed to look like Guzma and provides him coverage for his bug types against Flying types. Lokix and Yanmega were chosen to modernize / add variety to his team. 

Rose
His team in the games specialize in steel types so it's true here as well, he also retains his general main team but I swapped off Escavelier for Tinkaton (and Mawile per request) for better coverage against fighting types and also to get rid of him having 2 x4 fire weakness Pokemon. He was given Duraludon for the reason of being a steel type from his generation and electricity / energy thematic. Gholdengo and Aegislash were given to Rose also for the same reason except he runs an entire business across all of Galar (in other words, he is richer than Lysandre) but also the fact that he is a Steel specialist and Aegislash being a literal Sword and Shield on a guy from a game called Sword and Shield. Iron Crown was chosen as his paradox considering it's something he could build possibly, with Cobalion being a variant by request. Melmetal was chosen as the "boss" since we can't use Eternatus however it actually has box legend stats in PokeRogue due to the G-Max increasing its BST. Two G-Maxes might look awkward on him but then again, Melmetal is also technically a Gen 8 Pokemon and even had lore in the game concerning its G-Max, and Rose is from the game that the G-Max gimmick originated from. 

For the other leaders, I replaced Huntail / Gorebyss on Archie to modernize his team a bit further in fight 2, while also providing him new coverage against Grass types and also keeping the Swift Swim gimmick they were doing.

Cyrus lost Basculegion to give to Ghetsis, Gyarados was a really good lead Pokemon and I felt was better than Basculegion and Hisui Braviary was replaced with Gliscor to have a mon from his gen but also to keep the bat idea with Crobat.

Ghetsis leads with Hydreigon/Jugulis now in fight 2, Iron Moth was given to Lysandre and replaced with Volcarona (who can genuinely still hold its own weight.) Genesect is the new 3rd boss now, with Douse Drive and Protean passive to trip people up potentially, it was a Pokemon designed by Team Plasma and is made shiny to reference one of the BW movies. Basculegion replaces the Darmanitans to give him better coverage (Fire Darm was covered previously by Iron Moth and Ice Darm was covered by Kyurem) This also gives him a fighting immunity since his team was already susceptible to it before.

As mentioned earlier Lysandre lost the fairy paradoxes for Eternal Floette, and the Steel/Ghosts replaced with Iron Moth from Ghetsis, which I felt fit Lysandre considerably better one reason being the Team _Flare_ themeing and how he uses futuristic drones and equipment/technology in the games. Milotic was replaced to give to Lusamine as she also actually uses one and water was covered by Gyarados already, it was replaced with Florges to make the connection from Fight 1 to 2 with Eternal Floette. Heliolisk I felt was underwhelming and was replaced with Gen 6 Fossils but I agree there could be something better in that slot. His Gyarados was made shiny by the request of people as it also references the anime but I would say his shiny Gyarados is really iconic to have (and just differentiates it from other Gyarados)

Tempo-anon commented:
> Lusamine
Clefable was made the boss instead of her Mega Lopunny as it is her ace (and I'm following the logic you used with Cyrus and Weavile). There was discussion on what her mega should be, it was a choice between either Mega Lopunny or Mega Altaria which is a bit of a stretch since she gives one to Lillie in the anime. Lysandre lost Scream Tail / Flutter Mane to give to Lusamine as she also has the same warped perceptions of beauty (reasoning you gave for Lysandre) on top of the fact she actually has a Mismagius on her team and a Clefable that you could somewhat make a connection to Scream Tail for. I chose Eternal Floette as their replacement as it is also a Mythical Pokemon that retains the general fairy typing the paradoxes had while also being a really nice reference to XY and cool factor (the passive bonus of being a boss is also really nice for it). Espathra was given to modernize the team in fight 1 but it felt too fitting to not give to her in just pure design but also it was nice type coverage. She canonically owns a Magearna in the anime so it was chosen as the "boss" slot (Necrozma was intentionally not chosen for the same reason we can't give Rose Eternatus) and Nihilego is an obvious pick for her.

Is Clefable her actual ace in-game? I made Cyrus's Weavile the boss because it's quite clearly his ace (he's commonly pictured with it, it's his final Pokemon, his highest level Pokemon, the only one with an item) but for Lusamine it looks like her in-game team is all the same level. I'm still not too sold on Magearna though. There's something about it that just feels like it doesn't fit. Maybe something like a UB variant or Pheremosa since that's literally the UB that's based off her design? Her first team looks okay to me but for fight 2 I'm thinking something like
Scream Tail / Flutter Mane (boss)
Mega Lopunny
Nihilego (boss)
Pheremosa (boss)

> Guzma
Team 1 is mainly his in game team, Heracross was added to keep the type coverage when going into Team 2 with his Buzzwole. Galvantula and Vikavolt are for the same reason turning into Xurkitree. Kleavor was made a variation over his Scizor. Golisopod was chosen as the boss in round 1 as it is his ace and intentionally made to start first as it pairs with Emergency Exit + Regenerator passive. Ariados was added as a variation on Masquerain per request. (Typing this I realize he doesn't have an extra sub legend to make up for his lack of legend, could maybe give him Slither Wing over the Scizor slot) Xurkitree was chosen as the "boss" as it is designed to look like Guzma and provides him coverage for his bug types against Flying types. Lokix and Yanmega were chosen to modernize / add variety to his team.

No complaints about Guzma besides the requested addition of Ariados as a varation of Masquerain which you did. Looks good to me 👍 

> Rose
His team in the games specialize in steel types so it's true here as well, he also retains his general main team but I swapped off Escavelier for Tinkaton (and Mawile per request) for better coverage against fighting types and also to get rid of him having 2 x4 fire weakness Pokemon. He was given Duraludon for the reason of being a steel type from his generation and electricity / energy thematic. Gholdengo and Aegislash were given to Rose for the same reason you gave them to Lysandre initially, except he runs an entire business across all of Galar (in other words, he is richer than Lysandre) but also the fact that he is a Steel specialist and Aegislash being a literal Sword and Shield on a guy from a game called Sword and Shield. Iron Crown was chosen as his paradox considering it's something he could build possibly, with Cobalion being a variant by request. Melmetal was chosen as the "boss" since we can't use Eternatus however it actually has box legend stats in PokeRogue due to the G-Max increasing its BST. Two G-Maxes might look awkward on him but then again, Melmetal is also technically a Gen 8 Pokemon and even had lore in the game concerning its G-Max, and Rose is from the game that the G-Max gimmick originated from.

Aegislash being a literal sword and shield is actually 👌 flavor. Almost makes me want to say that it should be a guarantee instead of a possible variation with Gholdengo. No real complaints about the teams now, just that it's a bit odd that there's 2 g-maxes. Yes I know that in PokeRogue you can have 6 g-maxes on a team if you wanted but it's still a little bit jarring to see an opponent with two.

> For the other leaders, I replaced Huntail / Gorebyss on Archie to modernize his team a bit further in fight 2, while also providing him new coverage against Grass types and also keeping the Swift Swim gimmick they were doing.

sadge but sure

> Cyrus lost Basculegion to give to Ghetsis, Gyarados was a really good lead Pokemon and I felt was better than Basculegion and Hisui Braviary was replaced with Gliscor to have a mon from his gen but also to keep the bat idea with Crobat.

You replaced Overqwil with Gliscor as a variant to Crobat. HBraviary is still a variant for Honchkrow. This is okay I guess.

> Ghetsis leads with Hydreigon/Jugulis now in fight 2, Iron Moth was given to Lysandre and replaced with Volcarona (who can genuinely still hold its own weight.) Genesect is the new 3rd boss now, with Douse Drive and Protean passive to trip people up potentially, it was a Pokemon designed by Team Plasma and is made shiny to reference one of the BW movies. Basculegion replaces the Darmanitans to give him better coverage (Fire Darm was covered previously by Iron Moth and Ice Darm was covered by Kyurem) This also gives him a fighting immunity since his team was already susceptible to it before.

Keep in mind that the boss rematches use the ELITE_FOUR team comp which means the levels in the team are 150, 150, 153, 153, 153, 159 with the further back the Pokemon is the higher level. So Hydreigon should be one of the 153 slots unless you want to tweak the team comps for everything. I don't think making Genesect have a specific form index is necessary if it has Protean and you want to put Basculegion on the team anyway. Only 1 variant on the team makes the team feel too set in stone in my opinion. Durant and Darm were on the team to showcase some powerful gen 5 Pokemon. At least variant Volcarona with Slither Wing but a big part of the strength was that passive Iron Moth was a true menace.

> Lysandre

I still think Heliolisk was a cool addition to his team (Electric typing, potential dry skin, gen 6 representation) but the fossils for him don't really fit.

> Variations

Some of your new teams only have a single variation on the team. For reference the evil boss teams as is have:
Giovanni: 4, 1 
Maxie: 3, 3
Aqua: 3, 4
Cyrus: 4, 3
Ghetsis: 3, 3
Lysandre: 1, 3
Giovanni2 and Lysandre1 only having a single variation on the team is something I wish I can go back and tweak

Blitz425 commented:
> Is Clefable her actual ace in-game? I made Cyrus's Weavile the boss because it's quite clearly his ace (he's commonly pictured with it, it's his final Pokemon, his highest level Pokemon, the only one with an item) but for Lusamine it looks like her in-game team is all the same level. I'm still not too sold on Magearna though. There's something about it that just feels like it doesn't fit. Maybe something like a UB variant or Pheremosa since that's literally the UB that's based off her design? Her first team looks okay to me but for fight 2 I'm thinking something like Scream Tail / Flutter Mane (boss) Mega Lopunny Nihilego (boss) Pheremosa (boss)
> 
Flavor text it's implied her Clefable is her ace (was mostly going off anime for that one) but I have no problem with the proposed fight 2 changes. I'll gather more opinions on the Magearna choice but I'm worried about giving her two bosses that are fast sweepers in Flutter Mane and Pheromosa. I liked the idea of having it be a Magearna/Naganadel variant.

> You replaced Overqwil with Gliscor as a variant to Crobat. HBraviary is still a variant for Honchkrow. This is okay I guess.

Yeah that was my bad, I mixed those two up

> Keep in mind that the boss rematches use the ELITE_FOUR team comp which means the levels in the team are 150, 150, 153, 153, 153, 159 with the further back the Pokemon is the higher level. So Hydreigon should be one of the 153 slots unless you want to tweak the team comps for everything. I don't think making Genesect have a specific form index is necessary if it has Protean and you want to put Basculegion on the team anyway.

I wanted Genesect to intentionally throw people off with changing its offensive / defensive types mid battle. Douse Drive Techno Blast Protean removes its massive fire weakness (if it rolls the move even) so I didn't want him to reveal that card immediately by having it as a lead Pokemon. We could try to do the thing you mentioned where we could set those level values manually. 

> Only 1 variant on the team makes the team feel too set in stone in my opinion. Durant and Darm were on the team to showcase some powerful gen 5 Pokemon. At least variant Volcarona with Slither Wing but a big part of the strength was that passive Iron Moth was a true menace.

I definitely do not mind adding Slither Wing as a variant

> I still think Heliolisk was a cool addition to his team (Electric typing, potential dry skin, gen 6 representation) but the fossils for him don't really fit.

It's only in his first fight anyway so I won't make a fuss out of it, I'll revert it back to Heliolisk. 

> Giovanni2 and Lysandre1 only having a single variation on the team is something I wish I can go back and tweak

Honestly I used their lack of variation as justification when building teams, since I'm also changing a few of the other leaders I could include possible changes to them in this PR if you'd like.

Tempo-anon commented on src/data/trainer-config.ts:
```diff
@@ -1222,6 +1228,27 @@ export const trainerConfigs: TrainerConfigs = {
+    }),
+  [TrainerType.MACRO_GRUNT]: new TrainerConfig(++t).setHasGenders("Macro Grunt Female").setHasDouble("Macro Cosmos Trainers").setMoneyMultiplier(1.0).setEncounterBgm(TrainerType.PLASMA_GRUNT).setBattleBgm("battle_plasma_grunt").setMixedBattleBgm("battle_macro_grunt").setVictoryBgm("victory_team_plasma").setPartyTemplateFunc(scene => getEvilGruntPartyTemplate(scene))
+    .setSpeciesPools({
+      [TrainerPoolTier.COMMON]: [ Species.CUFANT, Species.GALAR_MEOWTH, Species.GALAR_STUNFISK, Species.KLINK, Species.ROOKIDEE, Species.MAGNEMITE],
+      [TrainerPoolTier.UNCOMMON]: [ Species.DRILBUR, Species.FERROSEED, Species.MAWILE, Species.BRONZOR, Species.FORRETRESS, Species.DURANT, Species.SKARMORY, Species.STEELIX],
+      [TrainerPoolTier.RARE]: [Species.PROBOPASS, Species.TINKATINK, Species.ORTHWORM, Species.ARON, Species.VAROOM, Species.HONEDGE, Species.LUCARIO],
+      [TrainerPoolTier.SUPER_RARE]: [Species.DURALUDON, Species.SCIZOR]
```
lmao I just noticed the entire pool is steel type. While that may be accurate to the game that's not quite the best for gameplay mechanics. Consider adding some nonsteel types that would also fit like Trubbish, Salandit (used by Oleana), Rotom (used by the pokeleague staff), and other early route common gen 8 mons

I also noticed that Steelix is noticeably missing from here.

Tempo-anon commented on src/data/trainer-config.ts:
```diff
@@ -1222,6 +1228,27 @@ export const trainerConfigs: TrainerConfigs = {
+    }),
+  [TrainerType.SKULL_GRUNT]: new TrainerConfig(++t).setHasGenders("Skull Grunt Female").setHasDouble("Team Skull Grunts").setMoneyMultiplier(1.0).setEncounterBgm(TrainerType.PLASMA_GRUNT).setBattleBgm("battle_plasma_grunt").setMixedBattleBgm("battle_skull_grunt").setVictoryBgm("victory_team_plasma").setPartyTemplateFunc(scene => getEvilGruntPartyTemplate(scene))
+    .setSpeciesPools({
+      [TrainerPoolTier.COMMON]: [ Species.SALANDIT, Species.ALOLA_RATTATA, Species.SCRAGGY, Species.KOFFING, Species.ALOLA_GRIMER, Species.MAREANIE, Species.YUNGOOS, Species.SPINARAK, Species.PARAS],
```
Since Guzzma is a bug specialist I wonder if more bugs can be added to the skull grunt pool. The general theme would then be gen 7 and bugs

CodeTappert commented on src/locales/de/bgm-name.ts:
```diff
@@ -79,11 +79,17 @@ export const bgmName: SimpleTranslationEntries = {
   "battle_galactic_grunt": "BDSP Team Galactic Battle",
   "battle_plasma_grunt": "SW Vs. Team Plasma Rüpel",
   "battle_flare_grunt": "XY Team Flare Battle",
+  "battle_aether_grunt": "SM Aether Foundation Battle",
+  "battle_skull_grunt": "SM Team Skull Battle",
+  "battle_macro_grunt": "SWSH Trainer Battle",
   "battle_rocket_boss": "USUM Giovanni Battle",
   "battle_aqua_magma_boss": "ORAS Archie & Maxie Battle",
   "battle_galactic_boss": "BDSP Cyrus Battle",
   "battle_plasma_boss": "B2W2 Ghetsis Battle",
   "battle_flare_boss": "XY Lysandre Battle",
+  "battle_aether_boss":  "SM Lusamine Battle",
+  "battle_skull_boss": "SM Guzma Battle",
+  "battle_macro_boss": "SWSH Rose Battle",
```
```suggestion
  "battle_aether_grunt": "SM Vs. Æther Foundation",
  "battle_skull_grunt": "SM Vs. Team Skull Rüpel",
  "battle_macro_grunt": "SWSH Vs. Trainer",
  "battle_rocket_boss": "USUM Vs, Giovanni",
  "battle_aqua_magma_boss": "ORAS Vs. Team Magma/Aqua Boss",
  "battle_galactic_boss": "BDSP Vs. Zyrus",
  "battle_plasma_boss": "S2W2 Vs, G-Cis",
  "battle_flare_boss": "XY Vs. Flordelis",
  "battle_aether_boss":  "SM Vs. Samantha",
  "battle_skull_boss": "SM Vs. Bromley",
  "battle_macro_boss": "SWSH Vs. Rose",
```

CodeTappert commented on src/locales/de/trainers.ts:
```diff
@@ -19,6 +19,9 @@ export const titles: SimpleTranslationEntries = {
   "galactic_boss": "Galaktik-Boss",
   "plasma_boss": "Weiser von Team Plasma", // This is on purpose, since "Ghetsis" is never mentioned as the boss of team plasma in the game but as "Weiser"
   "flare_boss": "Flare-Boss",
+  "aether_boss": "Aether President",
+  "skull_boss": "Team Skull Boss",
+  "macro_boss": "Macro Cosmos's",
```
```suggestion
  "aether_boss": "Æther-Präsidentin",
  "skull_boss": "Skull-Boss",
  "macro_boss": "Macro Cosmos",
```

CodeTappert commented on src/locales/de/trainers.ts:
```diff
@@ -143,6 +146,12 @@ export const trainerClasses: SimpleTranslationEntries = {
   "flare_grunt": "Rüpel von Team Flare",
   "flare_grunt_female": "Rüpel von Team Flare",
   "flare_grunts": "Rüpel von Team Flare",
+  "aether_grunt": "Aether Foundation Employee",
+  "aether_grunt_female": "Aether Foundation Employee",
+  "skull_grunt": "Team Skull Grunt",
+  "skull_grunt_female": "Team Skull Grunt",
+  "macro_grunt": "Macro Cosmos Trainer",
+  "macro_grunt_female": "Macro Cosmos Trainer",
```
```suggestion
  "aether_grunt": "Æther Foundation Personal",
  "aether_grunt_female": "Æther Foundation Personal",
  "skull_grunt": "Rüpel von Team Skull",
  "skull_grunt_female": "Team Skull Grunt",
  "macro_grunt": "Macro Cosmos",
  "macro_grunt_female": "Macro Cosmos",
```

CodeTappert replied:
Also. Macro Cosmos Trainers have a normal name like regular trainers. Could we implement that?

CodeTappert commented on src/locales/de/trainers.ts:
```diff
@@ -276,6 +285,9 @@ export const trainerNames: SimpleTranslationEntries = {
   "cyrus": "Zyrus",
   "ghetsis": "G-Cis",
   "lysandre": "Flordelis",
+  "lusamine": "Lusamine",
+  "guzma": "Guzma",
+  "rose": "Rose",
```
```suggestion
  "lusamine": "Samantha",
  "guzma": "Bromley",
  "rose": "Rose",
```

CodeTappert commented on src/locales/de/trainers.ts:
```diff
@@ -143,6 +146,12 @@ export const trainerClasses: SimpleTranslationEntries = {
+  "aether_grunt_female": "Æther Foundation Personal",
+  "skull_grunt": "Rüpel von Team Skull",
+  "skull_grunt_female": "Rüpel von Team Skull",
+  "macro_grunt": "Macro Cosmos",
+  "macro_grunt_female": "Macro Cosmos",
```
```suggestion
  "macro_grunt": "Angestellter von Macro Cosmos",
  "macro_grunt_female": "Angestellte von Macro Cosmos",
```

CodeTappert commented:
You missed to add the localization placeholders for trainers.ts in es,it and zh_TW

Fontbane commented on src/data/trainer-config.ts:
```diff
@@ -1628,9 +1655,9 @@ export const trainerConfigs: TrainerConfigs = {
       p.pokeball = PokeballType.MASTER_BALL;
```
Maxie: Galar Weezing might have anti synergy with Flygon if it rolls Misty Surge. Also if Maxie gets Crobat/Gliscor (fits him imo) Cyrus should probably get Crobat/Gengar in that slot instead.
Pelipper and Torkoal have their weather abilities in the second slot, not as Hidden abilities.

Fontbane commented on src/data/trainer-config.ts:
```diff
@@ -1524,7 +1551,7 @@ export const trainerConfigs: TrainerConfigs = {
       p.generateName();
     })),
   [TrainerType.ROCKET_BOSS_GIOVANNI_2]: new TrainerConfig(++t).setName("Giovanni").initForEvilTeamLeader("Rocket Boss", [], true).setMixedBattleBgm("battle_rocket_boss").setVictoryBgm("victory_team_plasma")
```
Giovanni: I'll maintain that he should have Honchkrow. He was one of the first to have it and it's quite fitting to him as a yakuza leader. This ground team fits his gym leader battle more (I plan to redo gym leader teams soon). I'd suggest to give him both Nidos or Ttar, and also give him a Honchkrow (could have a variant slot if you want).
Regarding battle 2: Tyranitar vs Iron Thorns is kind of a big variation in difficulty. I'd probably just do Iron Thorns as a boss here, and maybe put Ttar on his first team. Seismitoad can be a variant for Gastrodon. Rhyperior or a Nido could still be slotted in somewhere I think. Also- the Excadrill strat falls apart without sand, which could happen if Hippowdon is sent out while Thorns sand is still up. Could afford to put a variant or two in one of those slots since it doesn't really make sense to have a separate sand setter for essentially just one more sand abuser.

Fontbane commented on src/locales/ko/dialogue.ts:
```diff
@@ -563,6 +587,84 @@ export const PGMdialogue: DialogueTranslationEntries = {
+          $With Nihilego's power, I will show you how wrong you were to come here!`
+    },
+    "victory": {
+      1: `You hateful little Trainer! How dare you intrude upon this world that was meant for Nihilego and me?
```
ma'am this is a Tall Grass biome

Blitz425 commented on src/data/trainer-config.ts:
```diff
@@ -1192,7 +1198,7 @@ export const trainerConfigs: TrainerConfigs = {
       [TrainerPoolTier.UNCOMMON]: [Species.SOLROCK, Species.HIPPOPOTAS, Species.SANDACONDA, Species.PHANPY, Species.SWINUB, Species.GLIGAR],
       [TrainerPoolTier.RARE]: [Species.TRAPINCH, Species.HEATMOR],
-      [TrainerPoolTier.SUPER_RARE]: [Species.TURTONATOR, Species.CHARCADET]
+      [TrainerPoolTier.SUPER_RARE]: [Species.SCOVILLAIN, Species.CHARCADET]
```
I don't feel taking chances with stone evos as whatever coding we have for forcing an evolution likes to be inconsistent with those types of Pokemon, but the first Grunt fight is at Wave 35 regardless which logically the player could have gotten a stone at this point

Tempo-anon replied:
I think it's okay to leave the unevo'd forms here and in other places. If it's a problem with the evolution delay function then we should be fixing that instead.

Blitz425 commented:
> You missed to add the localization placeholders for trainers.ts in es,it and zh_TW

Those were missing the previous Evil Leaders so I don't think anyone went back to do them :S I'll leave that for someone else
